dexcom/alert.go

```diff
@@ -627,15 +629,17 @@ func (a *AlertSetting) validateUrgentLowSoon(validator structure.Validator) {
 }
 
 func ParseAlertScheduleSettingsTime(value string) (int, int, bool) {
-	if submatch := alertScheduleSettingsTimeExpression.FindStringSubmatch(value); len(submatch) != 3 {
+	timeFormat := "15:04"
```

Comment:
This could be handled simply by extending the original regex pattern below to accommodate the optional AM/PM indicator with zero or more spaces in between:

```
var alertScheduleSettingsTimeExpression = regexp.MustCompile("^([0-9]{1,2}):([0-9]{2})(?i:\\s*(AM|PM))?$")
```

plus the actual check would need to accommodate 3 or 4 matches:

```
if submatch := alert... ; len(submatch) >= 3 {
```

Reply:
I should note that that check won't catch non-sensical 12-hour clock values like "18:05 AM", but I think it's good enough for this use.

Reply:
I guess my assumption was the proposed solution covers both cases but also covers the fact we are actually dealing with times here by parsing them as times ... and will also cover the case you mention above of non sensical 12 hour times.

if regex we then also need to convert the 12hr into 24hr format for consistency 

thoughts?

Reply:
Ah, yea, fair point. Yea, let's stick with your parsing solution, even though we don't actually use the resulting time (sensical or not) for anything AFAIK.